### DISCUSSION TITLE
Extract the generated code in v1.0.9

### DISCUSSION THREAD

**Discussion by notmanas** [sinaptik-ai/pandas-ai#500] [Q&A] [answered]
Can I extracted the last code generated and store it in a variable?
Earlier, in version 0.8.1, I could just do this -

```
pandasai = PandasAI(llm=llm, save_charts=True, save_charts_path="tmp_chart_results")
generated_code = pandasai.last_code_executed
```

But now there is no such option. I have tried adding this code in pandasai/__init__.py:

```
def last_generated_code(self):
        return self._dl.last_code_generated
```

This does give me the result, but is there an inbuilt function for this? I'm not sure if editing the library code is the best option for me.

Thanks in advance!

**Comment by gventuri** [comment, marked as answer]
@notmanas thanks a lot for reporting. Although we try to maintain backward compatibility, the `PandasAI` class will be deprecated soon. We now use either `SmartDataframe` or `SmartDatalake`. Check out the docs: https://docs.pandas-ai.com/en/latest/

Those classes both support `last_generated_code`.

Let me know if that helps!

**Reply by notmanas** [reply]
@gventuri this is really helpful! Saved me a lot of time :)

**Reply by orbisDerekP** [reply]
If I'm not mistaken the correct callout is - `last_code_generated`, not  ``last_generated_code``.
